solr/core/src/java/org/apache/solr/api/ApiBag.java

```diff
@@ -301,19 +301,18 @@ public static SpecProvider constructSpec(PluginInfo info) {
     Object specObj = info == null ? null : info.attributes.get("spec");
     if (specObj == null) specObj = "emptySpec";
     if (specObj instanceof Map) {
-      @SuppressWarnings({"rawtypes"})
-      Map map = (Map) specObj;
+        assert false : "got a map from a string";
```

Comment:
info.attributes is a Map<String,String> so we should never be getting a specObj that is a Map. However, the code exists and I don't necessarily trust out type safety everywhere, so I wanted to give our tests a chance to prove me wrong.

Reply:
So there's actually something pretty horrible going on here, at PluginInfo line 141 (https://github.com/apache/solr/pull/165/files#diff-ed395ffeb39b23f571cff745a84a590312d3cceb3ef91ff9123e6cdcf4b7d14aL141) where we wrap the existing `Map<String,Object> initArgs` as `Map<String,String> attributes`. Then when we pull this spec through in ApiBag and lots of places around core initialization. It's a big issue but I think it needs to be handled separately from the rest of the raw types since I suspect there will be lots of changes as we figure out how to untangle it.

Reply:
Yes, definitely another issue -- https://issues.apache.org/jira/browse/SOLR-14696